src/components/requester.js

```diff
@@ -28,16 +30,38 @@ module.exports = class Requester extends Monitorable(Configurable(Component)) {
     }
 
     send(...args) {
-        if (args.length == 1 || typeof args[args.length - 1] != 'function') {
-            return new Promise((resolve, reject) => {
-                this.sock.send(...args, (err, res) => {
-                    if (err) return reject(err);
-                    resolve(res);
-                });
+        const cb = args[args.length - 1];
+        const hasCallback = 'function' == typeof cb;
```

Comment:
there can be an early return here —

```js
if (!hasCallback) return sockSend(args)
```

Reply:
also at this point args could be replaced with the cb omitted version, so that you don't have to do `sockSend(args.slice(0, -1))` later.

Reply:
Tried to implement these.